src/app/containers/Paragraph/index.jsx

```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import Paragraph from '../../components/Paragraph';
+import Blocks from '../Blocks';
+import fragment from '../Fragment';
+import { paragraphModelPropTypes } from '../../models/propTypes/paragraph';
+
+const componentsToRender = { fragment };
+
+const ParagraphContainer = ({ blocks }) => (
+  <Paragraph>
+    <Blocks blocks={blocks} componentsToRender={componentsToRender} />
```

Comment:
Why is this Paragraph container needed? 

I think inclusion of `Blocks` should happen at a higher container level.
Ideally, it should exist at the Article container level, where we define `componentsToRender` https://github.com/BBC-News/simorgh/blob/latest/src/app/containers/Article/index.jsx#L7 and pass that to the `Blocks` component.

Reply:
`Blocks` is now a generic component that renders objects inside a `blocks` array. It renders these objects using components that are passed to it.

On that, Paragraph is a container because it no longer just renders text into a styled component, but renders nested arrays (fragments inside blocks) into a string. It contains the configuration used to render these arrays via `Blocks`.

The current [`Text` container](/latest/src/app/containers/Text/index.jsx#L13-L14) does something very similar (`Text` can't be changed in this work), and a similar implementation has to be done in `Paragraph` as it has a similar data model (which is a blocks array that has to be iterated through).

I am unsure what you mean by `I think inclusion of Blocks should happen at a higher container level.`. Does that mean that the Article container should contain the references to what children components should pass to `Blocks` to render?

Finally, on your point about HOCs. Could you give a pseudo example of what you mean, and the benefits it would bring?

Reply:
We discussed this at length offline. There are several ways of approaching this.

- 1 Not have a paragraph container, and in the Text container handle looping through the multiple paragraphs and multiple fragments,  rendering the Fragment container. 
  
- 2 Not have a paragraph container, and have all of the above looping logic in the Articles container and rendering the Fragment container. 

- 3 Move the container mapping logic in the object  `componentsToRender = { fragment }` to the Articles container and use the React Context API to inform containers which other containers to render as child elements.

This last solution will mean our containers will be simpler, but when we pull them out into a component library, they will have to be used in conjunction with a wrapping Context component. This seems suboptimal.

Options 1 and 2 move the complexity out of separate containers and concentrate them in the Article and Text containers.

Option 2 would mean all our components and containers other than the Article container would be quite simple and easy to understand. This has the down side of making Article complex.

I would like the complexity due to our data feed to exist in a single place, at the start of the application, rather than spread across multiple containers.

Reply:
Separately to this PR, we should discuss the best approach for the containers. I'm happy to get this work merged in and we can talk about discussing this change at a later point.